### PR TITLE
Add to javadoc of getCheckpointMark

### PR DESCRIPTION
**Pull request by wtanaka**
Based on looking through various implementations of this method

### CODE REVIEW

**Comment by jbonofre** [comment]
It's also possible to return a kind of empty/void checkpoint mark like this:

```
    @Override
    public UnboundedSource.CheckpointMark getCheckpointMark() {
      return new UnboundedSource.CheckpointMark() {
        @Override
        public void finalizeCheckpoint() throws IOException {
          // nothing to do
        }
      };
    }
```
It's a kind of fake checkpoint but better then null IMHO.

**Comment by dhalperi** [comment]
R: @tgroh  @jbonofre

**Comment by tgroh** [comment]
I'm extremely leery of any Unbounded Source not requiring some form of checkpoint. To produce elements "exactly once", a source will almost certainly need to either:
* Resume from a point it's read in the stream, or
* Update the upstream state after elements are durably committed

Even if this is not the case, returning an empty checkpoint is of extremely low complexity, as @jbonofre suggested.

The only places we appear to return a `null` checkpoint mark within the Beam repositories are for testing sources and for `UnboundedSocketReader`, which does not have a well-defined output.

**Comment by wtanaka** [comment]
Updated based on example in https://github.com/apache/beam/pull/2406#pullrequestreview-30569397

**Comment by jbonofre** [comment]
retest this please

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java [hunk 1]
```diff
@@ -220,6 +220,20 @@ public boolean requiresDeduping() {
     /**
      * Returns a {@link CheckpointMark} representing the progress of this {@code UnboundedReader}.
      *
+     * <p>If this {@code UnboundedReader} does not support checkpoints, it may return a
+     * CheckpointMark which does nothing, like:
+     * <pre>{@code
```
Comment (jbonofre) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java]: This should be on two lines, like:

```
* CheckpointMark which does nothing, like:
*
* <pre>{@code
```

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java [hunk 2]
```diff
@@ -220,6 +220,20 @@ public boolean requiresDeduping() {
...
+     * @Override
```
Comment (jbonofre) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java]: This `@Override` can be "hooked" by javadoc.

**Comment by jbonofre** [comment]
By the way, you can test your javadoc change simply using:

```
mvn clean install -Prelease
```

**Comment by wtanaka** [comment]
@jbonofre done

**Pull request closed without merging**